spec/models/admin_dashboard_data_spec.rb

```diff
@@ -3,37 +3,129 @@
 require 'rails_helper'
 
 describe AdminDashboardData do
+  after do
+    AdminDashboardData.reset_problem_checks
+    Discourse.redis.flushdb
+  end
+
+  describe "#fetch_problems" do
+    describe "adding problem messages" do
+      it "adds the message and returns it when the problems are fetched" do
+        AdminDashboardData.add_problem_message("dashboard.bad_favicon_url")
+        problems = AdminDashboardData.fetch_problems.map(&:to_s)
+        expect(problems).to include(I18n.t("dashboard.bad_favicon_url", { base_path: Discourse.base_path }))
+      end
+
+      it "does not allow adding of arbitrary problem messages, they must exist in @problem_messages" do
+        AdminDashboardData.add_problem_message("errors.messages.invalid")
+        problems = AdminDashboardData.fetch_problems.map(&:to_s)
+        expect(problems).not_to include(I18n.t("errors.messages.invalid"))
+      end
+    end
+
+    describe "adding new checks" do
+      it 'calls the passed block' do
+        called = false
+        AdminDashboardData.add_problem_check do
+          called = true
+        end
+
+        AdminDashboardData.fetch_problems
+        expect(called).to eq(true)
+
+        AdminDashboardData.fetch_problems(check_force_https: true)
+        expect(called).to eq(true)
+      end
+
+      it 'calls the passed method' do
+        $test_AdminDashboardData_global = false
+        class AdminDashboardData
+          def my_test_method
+            $test_AdminDashboardData_global = true
+          end
+        end
+        AdminDashboardData.add_problem_check :my_test_method
 
-  describe "adding new checks" do
-    after do
-      AdminDashboardData.reset_problem_checks
+        AdminDashboardData.fetch_problems
+        expect($test_AdminDashboardData_global).to eq(true)
+        $test_AdminDashboardData_global = nil
+      end
```

Comment:
I think we end up leaking a global and the `my_test_method` into the `AdminDashboardData` class here. I think the following would work for what we're testing here. 

```
it 'calls the passed method' do
  klass = Class.new(AdminDashboardData) do
    cattr_reader :called

    def my_test_method
      @@called = true
    end
  end

  klass.add_problem_check :my_test_method

  klass.fetch_problems
  expect(klass.called).to eq(true)
end
```

Reply:
This spec and the weird `AdminDashboardData.fetch_problems(check_force_https: true)` one were already here I just moved them around. Not sure why it uses a global?? Might just delete them or rewrite them